src/csharp/Grpc.Core/Internal/NativeExtension.cs

```diff
@@ -176,17 +178,18 @@ private static string GetArchitectureString()
         // platform specific file name of the extension library
         private static string GetNativeLibraryFilename()
         {
+            string architecture = GetArchitectureString();
```

Comment:
You do the platform check twice, inside of `GetArchitectureString()` and here. Why not just move the architecture check here and return the hardcoded name of the library? The `String.Format` doesn't really gain much in this scenario.